### PR TITLE
[improve](group commit) Group commit support skip wal

### PR DESCRIPTION
**Pull request by mymeiyi**
## Proposed changes

Group commit support two modes:
* `group_commit = async_mode` write with wal, and returned once the block is consumed.
* `group_commit = sync_mode` write without wal, and returned after the internal load is finished.
* `group_commit = off_mode` disable group commit

This PR is based on https://github.com/apache/doris/pull/27720

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -652,8 +652,9 @@ struct TStreamLoadPutRequest {
     // only valid when file type is CSV
     52: optional i8 escape
     53: optional bool memtable_on_sink_node;
-    54: optional bool group_commit
+    54: optional bool group_commit // deprecated
     55: optional i32 stream_per_node;
+    56: optional string group_commit_type
```
Comment (dataroaring) [inline gensrc/thrift/FrontendService.thrift:657]: group_commit_mode

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run clickbench

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by mymeiyi** [comment]
run p0

**Pull request merged**